[mihirniyogi] iP

mihirniyogi opened the pull request:
# Bob

_Is that a bird? Is that a plane?_ 

No, it's 🔥<mark>BOB</mark>🔥 

- You may ask... why 'Bob'? 
- Why not some fancy name like 'Duke Royalty'?
- or 'iPhone XS Max Ultra+ Pro'?

because I believe in 

> Simplicity™️  

Much wow! So🆒COOL🆒! Bob The Task Manager is here to help make your task management 😎 SIMPLE 😎. 
Features include:

1. add todo, event or deadline
2. mark / unmark tasks
3. list all tasks
4. remove tasks
5. find tasks

Almost as simple as a notepad, like:
- [x] apples
- [x] bananas
- [ ] cherries

But almost as powerful as a full-on desktop application! For example, 'list' brings up all your tasks. 

&emsp;&emsp;Here are your tasks:
&emsp;&emsp;1. [E][ ] project meeting (from: Mar 17 2025 2:00pm to: Mar 17 2025 6:00pm)
&emsp;&emsp;2. [T][X] read book
&emsp;&emsp;3. [T][ ] return book
&emsp;&emsp;4. [D][ ] assignment (by: Feb 3 2025 11:59pm)

One day, Bob will become King Bob👑, see [here](https://www.youtube.com/watch?v=TjAg-8qqR3g).

---

For the nerds out there who want to edit Bob's precious source code to add more commands 🙄, you should edit `Parser::parse` and add another class that inherits from `Command`, as shown below:

```java
public class Parser {
    public static Command parse(String userInput) throws WrongCommandException, IOException {
        if (userInput.equals("bye")) {
            return new ExitCommand();
        }

        if (userInput.equals("list")) {
            return new ListCommand();
        }

        if (userInput.startsWith("mark")) {
            int number = Integer.parseInt(userInput.split(" ")[1]);
            return new MarkCommand(number);
        }

        // Add if-blocks here to catch more commands //
        // ...
    }    
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mihirniyogi committed:
Duke.java: Change Duke to Bob

mihirniyogi committed:
Add Bob's logo

mihirniyogi committed:
Add echo functionality

mihirniyogi committed:
Encapsulate printing into Helper class

mihirniyogi committed:
Improve encapsulation

mihirniyogi committed:
Enable task adding

mihirniyogi committed:
Enable task marking

mihirniyogi committed:
Enable task unmarking

mihirniyogi committed:
Refactor to remove TaskList.java

mihirniyogi committed:
Change Task to abstract class

mihirniyogi committed:
Create 3 task types

mihirniyogi committed:
Enable adding of 2 task types (T,D)

mihirniyogi committed:
Event class: fix toString method

mihirniyogi committed:
Enable task adding for event type

mihirniyogi committed:
Add test cases

mihirniyogi committed:
Create WrongCommandException class

mihirniyogi committed:
Add wrong input handling for task creation

mihirniyogi committed:
Add wrong input handling for mark, unmark

mihirniyogi committed:
Remove 'Exceptions' package, shift file up hierarchy

mihirniyogi committed:
Clean up code

mihirniyogi committed:
Add delete task functionality

mihirniyogi committed:
Install JUnit

mihirniyogi committed:
Create TaskList class to handle file IO + CRUD

mihirniyogi committed:
Add constructors to initialise tasks already having completion state

mihirniyogi committed:
Add read from file method

mihirniyogi committed:
Add convertLineToTask method

mihirniyogi committed:
Add GSON library

mihirniyogi committed:
Modify CSV method to work with JSON instead

mihirniyogi committed:
Revert "Add read from file method"

This reverts commit 01b1f2d5fdaf0dae601098cebf9992289537e382.

mihirniyogi committed:
Change test file to delete .csv before testing

mihirniyogi committed:
Add toCsv methods for convenience

mihirniyogi committed:
Do major refactor to use CSV as task storage

mihirniyogi committed:
Merge branch 'branch-Level-7'

mihirniyogi committed:
Add support for date time input

mihirniyogi committed:
Merge branch 'branch-Level-8'

mihirniyogi committed:
Abstract out print methods to Ui Class

mihirniyogi committed:
Bugfix: Correct datetime storage in CSV

mihirniyogi committed:
Abstract scanner and printTasks method to UI class

mihirniyogi committed:
Abstract out welcome message

mihirniyogi committed:
Abstract out file IO to Storage class

mihirniyogi committed:
Add trim to all commands being read

mihirniyogi committed:
Add abstract Command class

mihirniyogi committed:
Add necessary 'throw's to TaskList methods

mihirniyogi committed:
Add basic parse method

mihirniyogi committed:
Add ExitCommand

mihirniyogi committed:
Add ListCommand

mihirniyogi committed:
Add MarkCommand

mihirniyogi committed:
Add UnmarkCommand

mihirniyogi committed:
Add TodoCommand

mihirniyogi committed:
Add DeadlineCommand

mihirniyogi committed:
Add EventCommand

mihirniyogi committed:
Add DeleteCommand

mihirniyogi committed:
Switch to Command and Parser classes

mihirniyogi committed:
Merge branch 'branch-A-MoreOOP'

mihirniyogi committed:
Add all files into single 'bob' package

mihirniyogi committed:
Move all command files into bob.command pkg

mihirniyogi committed:
Move task-related files into bob.task pkg

mihirniyogi committed:
Move Parser into bob.parser pkg

mihirniyogi committed:
Move Storage into bob.storage pkg

mihirniyogi committed:
Move Ui class into bob.ui pkg

mihirniyogi committed:
Move Helper class into bob.util pkg

mihirniyogi committed:
Merge branch 'branch-A-Packages'

mihirniyogi committed:
Merge branch 'add-gradle-support'

mihirniyogi committed:
Delete JUnit jar file since gradle will handle in future

mihirniyogi committed:
Set entry point and jar name in gradle

mihirniyogi committed:
Add checkstyle in gradle

mihirniyogi committed:
Merge branch 'branch-A-Gradle'

mihirniyogi committed:
Add .vscode dir to gitignore

mihirniyogi committed:
Add Parser tests

mihirniyogi committed:
Add TodoCommandTest

mihirniyogi committed:
Add DeadlineCommandTest

mihirniyogi committed:
Add EventCommandTest

mihirniyogi committed:
Merge branch 'branch-A-JUnit'

mihirniyogi committed:
Add javadoc comments for all classes

mihirniyogi committed:
Add javadoc comments for most public methods and some non-trivial private methods

mihirniyogi committed:
Add find functionality

mihirniyogi committed:
Merge branch 'branch-A-JavaDoc'

mihirniyogi committed:
Merge branch 'branch-Level-9'

mihirniyogi committed:
Add javadoc for FindCommand class

mihirniyogi committed:
Match checkstyle suggestions and coding standards

mihirniyogi committed:
Merge branch 'branch-A-CodingStandard'

cyrolite commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    public static void main(String[] args) {
+
+        Ui ui = new Ui();
```
Consider declaring Ui as a final variable since it is not reassigned after initialization. 😄

mihirniyogi replied:
oh yea good point ! will do

cyrolite commented on src/main/java/bob/parser/Parser.java:
```diff
@@ -0,0 +1,127 @@
+        }
+
+        if (userInput.startsWith("mark")) {
+            int number = Integer.parseInt(userInput.split(" ")[1]);
```
It would be good if there was a way to check if the  split array has at least 2 elements to avoid `ArrayIndexOutOfBoundsException`.

mihirniyogi replied:
ah ok good point actually, this goes to show that my testing also needs improvement to catch this error 🤣

cyrolite commented on src/main/java/bob/command/Command.java:
I like how informative and clear the JavaDoc comments are. Good job!

cyrolite commented on src/main/java/bob/storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                .collect(Collectors.joining("\n"));
+        Files.write(FILE_PATH, lines.getBytes(StandardCharsets.UTF_8), StandardOpenOption.TRUNCATE_EXISTING);
+    }
+}
```
LGTM, excellent use of lambda expressions to help you with your application. I also like how tidy and organised the code is. Good job!

cyrolite commented on src/main/java/bob/task/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+        tasks.get(number - 1).unmark();
+        Storage.saveTasksToFile(tasks);
+    }
+}
```
These are some well-handled exception cases (e.g., IO errors, index out of bounds), coupled with appropriate messages. Well done!

mihirniyogi committed:
Add JavaFx depdendency

mihirniyogi committed:
Add Launcher.java

mihirniyogi committed:
Change entry point to launcher

mihirniyogi committed:
Add Main.java

mihirniyogi committed:
Make minimal echo GUI

mihirniyogi committed:
Add bob,user images to DialogBox

mihirniyogi committed:
Encapsulate images to DialogBox

mihirniyogi committed:
Add auto-scroll

mihirniyogi committed:
Do minor rename of method

mihirniyogi committed:
Add gradle tasks to differentiate between cli,gui

mihirniyogi committed:
Do major refactor of CLI

Command classes: Use ui instance to print out.

GUI and CLI have common logic.

GUI cannot use Command classes due to ui being used.

Let's
* extract out the common logic
* separate output from Command...and printing to CLI

mihirniyogi committed:
Comment out test files, errors to be fixed later

mihirniyogi committed:
Refactor package hierarchy, shift Bob and Ui to bob.cli

mihirniyogi committed:
Remove tab additions from Formatter class

mihirniyogi committed:
Remove unnecessary code - initialize is auto-called anyway

mihirniyogi committed:
Integrate gui with core logic

mihirniyogi committed:
Change height,width and prevent label from truncation

mihirniyogi committed:
Add stylistic changes

mihirniyogi committed:
Merge branch 'branch-Level-10'

mihirniyogi committed:
Add javadoc comments

mihirniyogi committed:
Add checkstyle edits

Partillay commented on src/main/java/bob/storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @return List of tasks.
+     */
+    public static List<Task> fetchTasksFromFile() {
```
Perhaps the method name can be shortened to fetchTasks() instead?

mihirniyogi replied:
good point👍

Partillay reviewed (commented):
The import commands are arranged nicely and overall code is really neat!

mihirniyogi committed:
Enable assertions via gradle

mihirniyogi committed:
Add assertions

mihirniyogi committed:
Fix bug: when input is empty, do not process; caught due to assertion

mihirniyogi committed:
Add comments to increase readability of Parser

mihirniyogi committed:
Change indentation to increase readability

mihirniyogi committed:
Merge pull request #1 from mihirniyogi/branch-A-Assertions

Merge branch-A-Assertions

mihirniyogi committed:
Merge pull request #2 from mihirniyogi/branch-A-CodeQuality

Merge branch-A-CodeQuality

mihirniyogi committed:
Add yml file for CI

mihirniyogi committed:
Update yaml to make it cleaner

mihirniyogi committed:
Fix checkstyle errors

mihirniyogi committed:
Exclude checkstyleTest from CI

mihirniyogi committed:
Delete text ui tests

mihirniyogi committed:
Modify delete command for multiple deletions

mihirniyogi committed:
Add Exit message constant and return empty string

mihirniyogi committed:
Abstract core logic between cli and gui out

mihirniyogi committed:
Merge pull request #3 from mihirniyogi/branch-MassOps

Merge branch-MassOps

mihirniyogi committed:
Change variable 'completed' to 'isCompleted'

mihirniyogi committed:
Refactor to shorten parse method

mihirniyogi committed:
Remove invalid assertion

mihirniyogi committed:
Fix NullPointerException in Parser

mihirniyogi committed:
Merge pull request #5 from mihirniyogi/branch-A-CodeQuality

Merge branch-A-CodeQuality

mihirniyogi committed:
Refactor error handling

Previously: some errors are re-processed to be WrongCommandException.

Change: Remove try-catch blocks everywhere except bob.core.Bob.

Now: all errors are thrown and propagated until bob.core.Bob.

mihirniyogi committed:
Merge pull request #6 from mihirniyogi/branch-A-MoreErrorHandling

Refactor error handling

mihirniyogi committed:
Add product screenshot

mihirniyogi committed:
Add user guide

mihirniyogi committed:
Change filepath to non-final for testing, add necessary methods

mihirniyogi committed:
Fix regression errors with MassOps

mihirniyogi committed:
Change command tests to work with GUI

mihirniyogi committed:
Add more valid and invalid input testing

mihirniyogi committed:
Edit gradle to now include checkstyleTest (previously excluded)

mihirniyogi committed:
Add Locale.US to standardise AM/PM capitalisation

mihirniyogi committed:
Merge pull request #7 from mihirniyogi/branch-A-MoreTesting

Merge branch-A-MoreTesting

mihirniyogi committed:
Change README
